api/v1beta1/keystoneapi.go

```diff
@@ -27,13 +28,41 @@ import (
 	"github.com/openstack-k8s-operators/lib-common/modules/common/secret"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/tls"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
 
 	openstack "github.com/openstack-k8s-operators/lib-common/modules/openstack"
 	appsv1 "k8s.io/api/apps/v1"
 	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// KeystoneAPIStatusChangedPredicate - primary purpose is to return true if
+// the KeystoneAPI Status.APIEndpoints has changed.
+// In addition also returns true if it gets deleted. Used by service operators
+// to watch KeystoneAPI endpoint they depend on.
+var KeystoneAPIStatusChangedPredicate = predicate.Funcs{
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		if e.ObjectOld == nil || e.ObjectNew == nil {
+			return false
+		}
+		oldPod, okOld := e.ObjectOld.(*KeystoneAPI)
+		newPod, okNew := e.ObjectNew.(*KeystoneAPI)
+
+		if !okOld || !okNew {
+			return false // Not a keystonev1.KeystoneEndpoint, or cast error
```

Comment:
@stuggi just noticed the comment here references Endpoints. It should say KeystoneAPI right?

Reply:
If you check bellow the deepequal, it is checking the keystoneapi endpoints from the status